File: tests/fixtures/sample-project/src/services/wallet.rb
require_relative '../utils/validator'
require_relative '../utils/logger'

# WalletService handles wallet operations
class WalletService
  def initialize
    @logger = Logger.new
  end

  def create_wallet(user_id)
    @logger.info("Creating wallet")
    validate_address(user_id)
    persist_wallet(user_id)
  end

  def get_balance(wallet_id)
    @logger.info("Getting balance")
    fetch_balance(wallet_id)
  end

  private

  def persist_wallet(user_id)
    { id: "1", user_id: user_id, balance: 0 }
  end

  def fetch_balance(wallet_id)
    100.0
  end
end

File: tests/fixtures/sample-project/src/services/wallet.ts
import { validateAddress } from '../utils/validator';
import { Logger } from '../utils/logger';

export class WalletService {
  private logger: Logger;

  constructor() {
    this.logger = new Logger();
  }

  async createWallet(userId: string): Promise<Wallet> {
    this.logger.info('Creating wallet');
    validateAddress(userId);
    return this.persistWallet(userId);
  }

  async getBalance(walletId: string): Promise<number> {
    this.logger.info('Getting balance');
    return this.fetchBalance(walletId);
  }

  private async persistWallet(userId: string): Promise<Wallet> {
    return { id: '1', userId, balance: 0 };
  }

  private async fetchBalance(walletId: string): Promise<number> {
    return 100;
  }
}

interface Wallet {
  id: string;
  userId: string;
  balance: number;
}
